classes/framework/table_base.php

```diff
@@ -292,24 +292,18 @@ public final function save($sneakily = false) {
      * @throws \coding_exception
      * @return string
      */
-    public static final function get_table_name() {
-
-        if (empty(static::$table_name)) {
-            $class_name = get_called_class(); // 'mod_coursework\models\deadline_extension'
-            $pieces = explode('\\', $class_name); // 'mod_coursework', 'models', 'deadline_extension'
-            $table_name = end($pieces); // 'deadline_extension'
-            //todo this may need further attention but fixing the one issue identified so far.
-            if ($table_name == 'personal_deadline') {
-                // Table personal_deadline does not exist but coursework_person_deadline does.
-                $table_name = 'coursework_person_deadline';
-            } else {
-                $table_name .= 's'; // 'deadline_extensions'
-            }
+    final public static function get_table_name() {
+
+        if (empty(static::$tablename)) {
+            $classname = get_called_class(); // 'mod_coursework\models\deadline_extension'
+            $pieces = explode('\\', $classname); // 'mod_coursework', 'models', 'deadline_extension'
+            $tablename = end($pieces); // 'deadline_extension'
+            $tablename .= 's'; // 'deadline_extensions'
```

Comment:
reverting you're fix from previous MR. intentional?

Reply:
Yes intentional.  A correction I made elsewhere rendered the previous fix redundant